### PR TITLE
Add APP_DB to SAI mapping

### PR DESCRIPTION
**Pull request by marian-pritsak**

### CODE REVIEW

File: documentation/general/design/dash-sonic-hld.md [hunk 1]
```diff
@@ -443,10 +444,84 @@ SONiC for DASH shall have a lite swss initialization without the heavy-lift of e
 |                          | SAI_SWITCH_ATTR_VXLAN_DEFAULT_PORT  |  
 |                          | SAI_SWITCH_ATTR_VXLAN_DEFAULT_ROUTER_MAC |  
 
-### 3.3.4 Underlay Routing
+### 3.3.4 DASH orchagent (Overlay)
```
Comment (prsunny) [inline documentation/general/design/dash-sonic-hld.md:447]: Please move this to 3.3.2 which is currently the section for Overlay

File: documentation/general/design/dash-sonic-hld.md [hunk 2]
```diff
@@ -443,10 +444,84 @@ SONiC for DASH shall have a lite swss initialization without the heavy-lift of e
...
+| DASH_ACL_GROUP      |               |                  |                                                   |                                                 |
+|                     | group_id      |                  |                                                   |                                                 |
+|                     |               | ip_version       |                                                   |                                                 |
+| DASH_ACL_RULE       |               |                  |                                                   |                                                 |
+|                     | group_id      |                  |                                                   |                                                 |
+|                     | rule_num      |                  |                                                   | Why not put priority in the key?                |
+|                     |               | priority         |                                                   | Missing from SAI                                |
```
Comment (prsunny) [inline documentation/general/design/dash-sonic-hld.md:487]: Priority can possibly change. So its not possible to have it as key.

**Comment by marian-pritsak** [comment]
List of unused attributes so far:
* SAI_DASH_ACL_ATTR_COUNTER_ID
* SAI_INBOUND_ENI_TO_VM_ENTRY_ATTR_INBOUND_VM_ID
* SAI_INBOUND_ENI_TO_VM_ENTRY_ATTR_INBOUND_VM_ID
* SAI_INBOUND_VM_ATTR_UNDERLAY_DIP
* SAI_INBOUND_VM_ATTR_VNI
* SAI_INBOUND_ROUTING_ENTRY_ATTR_ACTION
* SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_USE_DST_VNI
* SAI_PA_VALIDATION_ENTRY_ATTR_ACTION
* SAI_DIRECTION_LOOKUP_ENTRY_ATTR_ACTION
* SAI_ENI_ETHER_ADDRESS_MAP_ENTRY_ATTR_ENI_ID

File: documentation/general/design/dash-sonic-hld.md [hunk 3]
```diff
@@ -443,10 +444,84 @@ SONiC for DASH shall have a lite swss initialization without the heavy-lift of e
 |                          | SAI_SWITCH_ATTR_VXLAN_DEFAULT_PORT  |  
 |                          | SAI_SWITCH_ATTR_VXLAN_DEFAULT_ROUTER_MAC |  
 
-### 3.3.4 Underlay Routing
+### 3.3.4 DASH orchagent (Overlay)
+
+|     APP_DB Table    |      Key      |       Field      |                 SAI Attributes                    |                     Comment                     |
+|---------------------|---------------|------------------|---------------------------------------------------|-------------------------------------------------|
+| DASH_APPLIANCE      |               |                  |                                                   |                                                 |
+|                     | appliance_id  |                  |                                                   | Only one?                                       |
```
[the remark this answers is not in the document]
Comment (marian-pritsak) [reply documentation/general/design/dash-sonic-hld.md]: How will the schema support two VIPs - v4 and v6?

**Comment by prsunny** [comment]
@prabhataravind to review

File: documentation/general/design/dash-sonic-hld.md [hunk 4]
```diff
@@ -341,6 +342,75 @@ mac_address              = MAC address as string     ; Inner dst mac
...
+| DASH_APPLIANCE      |               |                  |                                                   |                                                 |
+|                     | appliance_id  |                  |                                                   |                                                 |
+|                     |               | sip              | sai_vip_entry_t.vip                               |                                                 |
+|                     |               | vm_vni           | sai_direction_lookup_entry_t.VNI                  |                                                 |
+| DASH_VNET           |               |                  |                                                   |                                                 |
+|                     | vnet_name     |                  |                                                   |                                                 |
+|                     |               | vxlan_tunnel     |                                                   | VxLAN tunnel won't be used                      |
```
Comment (prsunny) [inline documentation/general/design/dash-sonic-hld.md:355]: Please remove vxlan_tunnel field.

File: documentation/general/design/dash-sonic-hld.md [hunk 5]
```diff
@@ -341,6 +342,75 @@ mac_address              = MAC address as string     ; Inner dst mac
...
+|                     |               | eni_id**         | sai_outbound_eni_to_vni_entry_t.ENI               |                                                 |
+|                     |               | qos              |                                                   |                                                 |
+|                     |               | vnet**           | SAI_OUTBOUND_ENI_TO_VNI_ENTRY_ATTR_VNI            | VNI value taken from DASH_VNET table            |
+| DASH_ACL_V4_IN      |               |                  | SAI_DASH_ACL_ATTR_STAGE                           | SAI_DASH_ACL_STAGE_INBOUND*                     |
+|                     | eni           |                  | SAI_DASH_ACL_ATTR_ENI_ID                          |                                                 |
+|                     |               | stage            | SAI_DASH_ACL_ATTR_STAGE                           | SAI_DASH_ACL_STAGE_INBOUND_STAGE{{stage}}       |
+|                     |               | acl_group_id     |                                                   |                                                 |
```
Comment (prsunny) [inline documentation/general/design/dash-sonic-hld.md:376]: Add mapping for ACL group

Comment (marian-pritsak) [reply documentation/general/design/dash-sonic-hld.md]: Done

**Review by prabhataravind — APPROVED** [review]
Looks good.

**Pull request merged**